### ISSUE TITLE
Consider removing docutils from dependencies

### ISSUE THREAD

**Issue by rudyryk** [boto/botocore#900]
As far as I can see, `docutils` package is only required for developing `botocore` itself and can be left out for `botocore` users.

I'd suggest considering to remove `docutils` from package dependencies as it adds significant time overhead to the installation process.

**Comment by Fale** [comment]
I did not noticed this one but I opened #933 for the same reason

**Comment by JordonPhillips** [comment]
Just to copy over some info from the linked PR:

> I think we'll need to first move textwriter.py up to the CLI (where it's used) before we can merge this.

**Comment by amw** [comment]
I was also surprised finding documentation utility as a direct dependency of a SDK package. I think the issue is significant, because `docutils` uses non-trivial license. Parts of it are even covered by GPL3.

http://docutils.sourceforge.net/COPYING.html

**Comment by yan12125** [comment]
Besides aws-cli, aws-shell also uses `botocore.docs.bcdoc.textwriter` [1]. As aws-shell appears to have been abandoned (last commit is from 2018), I don't think it is a blocker for moving the textwriter module from botocore to aws-cli, as implemented in [2].

[1] https://github.com/awslabs/aws-shell/blob/0.2.1/awsshell/makeindex.py#L8
[2] https://github.com/aws/aws-cli/pull/4917

**Comment by kapilt** [comment]
> I was also surprised finding documentation utility as a direct dependency of a SDK package. I think the issue is significant, because `docutils` uses non-trivial license. Parts of it are even covered by GPL3.
> 
> http://docutils.sourceforge.net/COPYING.html

the gpl part is just the emacs integration which is not part of the distributed assets on pypi, ie its not part of the package, its just present in their repo.

**Comment by nateprewitt** [comment]
This should now be resolved in botocore 1.18.0.
